config/samples/function-sample-with-host-based-route.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: core.openfunction.io/v1beta1
+kind: Function
+metadata:
+  name: function-sample-with-host-based-route
+spec:
+  version: "v1.0.0"
+  image: "openfunctiondev/v1beta1-http:latest"
+  port: 8080 # default to 8080
+  serving:
+    runtime: knative
```

Comment:
the filename is host-based route, but actually, the route is path based

Reply:
the route is host-based:
restrict path route is host-based and restrict hostname route is path-based.

Reply:
This is host based route for `function-sample-with-host-based-route.yaml` ?
![image](https://user-images.githubusercontent.com/18525465/180741623-3aa475ee-82ab-4614-87a4-592550929c50.png)

Reply:
yeah，If the route contains path, then the path is no longer generated according to the `pathTemplate`, only hostnames will be generated according to the `hostTemplate`, so it is host-based, e.g:
`http://function-sample.default.ofn.io/`

Reply:
ok, understood.
we might need to adjust the logic later because it's a bit weird to define a path like `/` while a `host` is needed to access a function

Reply:
when neither `hostname` nor `path` is defined, the default is `host-based`, so in fact, users can use `host-based` routing without defining a `path`.